homeassistant/components/sensor/teamspeak.py

```diff
@@ -0,0 +1,112 @@
+"""
+homeassistant.components.sensor.teamspeak
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Monitors a teamspeak server
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.teamspeak/
+"""
+import logging
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import convert
+
+REQUIREMENTS = ['ts3==0.7.1']
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Teamspeak Server'
+ICON = 'mdi:microphone'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Teamspeak sensor. """
+    name = config.get('name', DEFAULT_NAME)
+    host = config.get('host')
+    username = config.get('username')
+    password = config.get('password')
+    virtualserver_id = convert(config.get('virtualserver_id'), int, 0)
+
+    if host is None:
+        _LOGGER.error('No teamspeak host specified')
+        return
+    elif username is None:
+        _LOGGER.error('No teamspeak username specified')
+        return
+    elif password is None:
+        _LOGGER.error('No teamspeak password specified')
+        return
+
+    add_devices([
+        TeamspeakSensor(name, host, username, password, virtualserver_id)
+    ])
+
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+class TeamspeakSensor(Entity):
+    """ A Teamspeak Server sensor. """
+
+    def __init__(self, name, host, username, password, virtual_server_id):
+        self._name = name
+        self._state = None
+        self._unit_of_measurement = 'Users'
+        self._ts3conn = None
+
+        self._host = host
+        self._username = username
+        self._password = password
+        self._virtual_server_id = virtual_server_id
+
+        self.connect()
+        self.update()
+
+    @property
+    def name(self):
+        """ The name of the sensor. """
+        return self._name
```

Comment:
Good idea,ty; I've changed it now so that the user can still set it in config but it will default to the server name. I've done this in the init function because it seems the name only has to be set at initialization? Or is this bad practice?